Sources/CredentialsFacebook/TypeSafeFacebook.swift

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright IBM Corporation 2018
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import Foundation
+import KituraNet
+import Credentials
+import LoggerAPI
+import TypeDecoder
+
+/// A protocol that defines common attributes of Facebook authentication methods (token
```

Comment:
This sentence makes sense until we get to ( and then it stops making sense to me.

Reply:
This was meant to represent a list of the types of Facebook authentication we could provide.  But it probably would be better to get rid of it.